packages/viewer/src/components/viewer-transform-widget/util.ts

```diff
@@ -99,22 +100,40 @@ export function computeInputTransform(
 
   switch (identifier) {
     case 'x-translate':
-      return Matrix4.makeTranslation(Vector3.create(position(), 0, 0));
+      return appliedToCurrent(
```

Comment:
nit: Should the appliedToCurrent be invoked one time and the underlying translation that changes be moved to another function? That way the wrapped function can be called once. This is probably more of a stylistic preference. So take it or leave it. :) 



Something like:


```typescript
function computedTranslation(identifier) {
  switch (identifier)
    case 'x-translate':
   return Matrix4.makeTranslation(Vector3.create(position(), 0, 0))
   
   ... case 'y-translate': ...
   }
```

Then this function can just return  something like:

```typescript
return appliedToCurrent( current, computedTranslation(identifier));
```

Reply:
Yeah, good thought. I'll take a look at reorganizing this method. Might be some other reductions in logic that'd be possible in these utilities. I'll take a look